commit dc1631b7dc508cb67fa6983007ddba3a2314d6bf
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Tue May 27 20:28:56 2014 +0200

    Add missing memory clobber
    
    * i386/i386/xen.h (mb, rmb, wmb): Add memory clobber.

diff --git a/i386/i386/xen.h b/i386/i386/xen.h
index 5bdaf0b..638d671 100644
--- a/i386/i386/xen.h
+++ b/i386/i386/xen.h
@@ -33,7 +33,7 @@
 
 /* TODO: this should be moved in appropriate non-Xen place.  */
 #define barrier() __asm__ __volatile__ ("": : :"memory")
-#define mb() __asm__ __volatile__("lock; addl $0,0(%esp)")
+#define mb() __asm__ __volatile__("lock; addl $0,0(%%esp)":::"memory")
 #define rmb() mb()
 #define wmb() mb()
 MACH_INLINE unsigned long xchgl(volatile unsigned long *ptr, unsigned long x)
